### PR TITLE
Add RPM spec file

### PR DESCRIPTION
**Pull request by GawainLynch**
Wasn't sure of the project's desired location, but will happily rebase with a different location after review.
* ~~Tested on RHEL/CentOS 6 & 7~~
* Tested on RHEL/CentOS 7
* Tested on Fedora 21

### CODE REVIEW

**Comment by jrobeson** [comment]
It would be cool to have this officially in fedora/EPEL.   In the near term we could have builds on https://copr.fedoraproject.org/ though.

**Comment by am11** [comment]
I think it does not compile with GCC 4.6 (or even 4.4.7, which was released after 4.6) right away, unless you apply [this kind of a patch](https://github.com/am11/libsass/commit/5138b02d8de139d3b26f7d780fe42ec728419c06) (originally provided by @mgreter) for libsass v3.1.0. I testing building node-sass with many versions of GCC (4.4.7, the manually build 4.5.2 and 4.5.4, 4.6 and 4.9) but since our requirement for node-sass was to support CentOS 5 (basically the specific glibc version that it was released with) and above, I stopped and applied the aforementioned patch to compile with 4.6. But I think version 4.7.x might be the bare minimum, as `CentOS 6.{don't-remember}` was able to build libsass as is, with its default dev-toolkit installation.

:+1: for making official RPM package of libsass! Any ETA on this?

@jrobeson, can we build on Copr today? It seems like it is active and community is engaged building packages using copr.

**Comment by jrobeson** [comment]
you just need a FAS account and the copr package and you can get started pretty quickly.

**Comment by mgreter** [comment]
@xzyfer I'm fine with merging this, although I didn't test it!?

**Comment by xzyfer** [comment]
@GawainLynch I believe gcc 4.7 is the minimum required. Can confirm you had gcc 4.6 on the environments you tested this on?

Otherwise LGTM.

**Comment by GawainLynch** [comment]
Derp! The CentOS 6 build machine I used a patch similar to the one that @am11 linked! I just retried on a vanilla install (gcc-* 4.4.7), and it indeed won't build. 

As for CentOS 7, that comes with GCC 4.8. I *think* I put GCC 4.6 as the minimum from something in the docs somewhere, and don't have a 4.6 install handy to test.

Should I just bump this to 4.7?

**Comment by xzyfer** [comment]
I thought the 4.6 was the minimum but according @am11 it's 4.7. Update that
field and it's good to go.

**Comment by GawainLynch** [comment]
Updated and rebased on `master`.

File: libsass.spec [hunk 1]
```diff
@@ -0,0 +1,66 @@
+Name:           libsass
+Version:        3.1.0
```
Comment (xzyfer) [inline libsass.spec]: Does this need to be 3.2.0? That's when this will land.

File: libsass.spec [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+%files devel
+%doc
+%{_includedir}/*
+%{_libdir}/*.so
+%{_libdir}/pkgconfig/*.pc
+
+
+%changelog
+* Tue Feb 10 2015 Gawain Lynch <gawain.lynch@gmail.com> - 3.1.0-1
```
Comment (xzyfer) [inline libsass.spec]: `3.1.0-1` what does this version number represent?

Comment (GawainLynch) [inline libsass.spec]: With RPM packages the `-n` is the packaging number that is incremented for each build of *that specific* version.

Comment (xzyfer) [inline libsass.spec]: Should this be 3.2.0 since that's the stable release this will land in?

**Comment by xzyfer** [comment]
@mgreter can we automate these version numbers like the make file?

**Comment by jrobeson** [comment]
I've never tried the CPack generated rpm files, but maybe those are worth a try? Then it could be part of the build process (once the CMake build system PR is finished)

**Pull request merged**